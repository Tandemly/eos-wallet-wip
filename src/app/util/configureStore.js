import { createStore, compose, combineReducers } from 'redux';
import { persistStore, autoRehydrate } from 'redux-persist';
import { reducer as reduxFormReducer } from 'redux-form';
import { reducer as account } from 'containers/Balance/reducer';
import { reducer as app } from 'containers/App/reducer';
import { reducer as login } from 'containers/Login/reducer';
import { reducer as notification } from 'containers/Notifications/reducer';
import { reducer as transactions } from 'containers/Transactions/reducer';
import { reducer as users } from 'containers/Users/reducer';
import middlewares from "middleware";


const reducers = combineReducers({
  app,
  account,
  form: reduxFormReducer,
  login,
  notification,
  transactions,
  users,
});

function configureStoreAsync() {
  return new Promise((resolve, reject) => {
    try {
      const store = 
        createStore(
          reducers,
          undefined,
          compose(
            autoRehydrate(),
            middlewares
          ),
        );

      persistStore(
        store,
        { },
        () => resolve(store)
      );
    } catch (e) {
      reject(e);
    }
  });
}

// NOTE sans rehydrate / local storage
function configureStore(preloadState) {
  return createStore(
    reducers,
    preloadState,
    middlewares,
  );
}

export {
  configureStoreAsync,
  configureStore
};
