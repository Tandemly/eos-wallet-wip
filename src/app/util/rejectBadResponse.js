const rejectBadResponse = response => (
  response.ok ? response : Promise.reject(response)
);

export default rejectBadResponse;
