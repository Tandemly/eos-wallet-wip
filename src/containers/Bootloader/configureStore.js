import { createStore, compose } from "redux";
import { persistStore, autoRehydrate } from "redux-persist";
import middlewares from "./middleware";
// import { reducers } from "./containers";

function configureStore() {
  return new Promise((resolve, reject) => {
    try {
      const store = createStore(
        (state = {}, action) => state,
        undefined,
        compose(autoRehydrate(), middlewares)
      );

      persistStore(store, {}, () => resolve(store));
    } catch (e) {
      reject(e);
    }
  });
}

export default configureStore;
