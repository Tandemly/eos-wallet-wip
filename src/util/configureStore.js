import { createStore, compose, combineReducers, applyMiddleware } from "redux";
import { persistStore, autoRehydrate } from "redux-persist";
import { reducer as reduxFormReducer } from "redux-form";
import { routerMiddleware } from "react-router-redux";
import { routerReducer } from "react-router-redux";
import accountReducer from "redux-modules/eos-account";
import { reducer as app } from "containers/App/reducer";
import { reducer as login } from "redux-modules/login/reducer";
import { reducer as notification } from "redux-modules/notifications/reducer";
import { reducer as transactions } from "redux-modules/transactions/reducer";
import { reducer as users } from "redux-modules/users/reducer";
import { createMemoryHistory } from "history";
import middlewares from "middleware";

const reducers = combineReducers({
  app,
  'eos-account': accountReducer,
  form: reduxFormReducer,
  login,
  notification,
  transactions,
  users,
  routing: routerReducer
});

function configureStoreAsync(history) {
  /* eslint-disable-next-line no-underscore-dangle */
  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const enhancers = composeEnhancers(
    applyMiddleware(...middlewares, routerMiddleware(history))
  );

  return new Promise((resolve, reject) => {
    try {
      const store = createStore(reducers, compose(autoRehydrate(), enhancers));

      persistStore(store, {}, () => resolve(store));
    } catch (e) {
      reject(e);
    }
  });
}

// NOTE sans rehydrate / local storage
function configureStore(preloadState) {
  const history = createMemoryHistory();

  return createStore(
    reducers,
    preloadState,
    compose(applyMiddleware(...middlewares, routerMiddleware(history)))
  );
}

export { configureStoreAsync, configureStore };
