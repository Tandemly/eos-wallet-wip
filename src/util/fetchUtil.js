/* global fetch */

const apiHost = process.env.REACT_APP_API_URI;
const apiKey = process.env.REACT_APP_API_KEY;
const defaultOptions = {
  method: "GET",
  headers: {
    "Content-type": "application/json",
    Authorization: `Bearer ${apiKey}`
  }
};

export const apiRequest = async (path: string, options = {}) => {
  let resp;
  try {
    resp = await fetch(`${apiHost}${path}`, {
      ...defaultOptions,
      ...options
    });
    // just grab as text (rather than check response headers for content type)
    let text = await resp.text();
    try {
      // attempt to parse as JSON
      let json = JSON.parse(text);
      return json;
    } catch (err) {
      // Not json, format and return
      return { status: resp.status, text: resp.statusText, message: text };
    }
  } catch (err) {
    return { status: resp.status, text: resp.statusText };
  }
};
