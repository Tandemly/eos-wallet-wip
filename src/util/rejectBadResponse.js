const rejectBadResponse = response => {
  console.log("reject?", response.ok, response.status);
  return response.ok ? response : Promise.reject(response);
};

export default rejectBadResponse;
