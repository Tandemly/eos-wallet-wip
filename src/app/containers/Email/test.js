import * as React from "react";
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';
import { Provider } from 'react-redux';
import { configureStore } from 'util/configureStore';
import EmailContainer from './';

describe('<EmailContainer />', () => {
  it('renders without crashing', () => {
    const staticContext = {};
    const store = configureStore();
    const tree = renderer.create(
      <StaticRouter location="/" context={staticContext}>
        <Provider store={store}>
          <EmailContainer />
        </Provider>
      </StaticRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
