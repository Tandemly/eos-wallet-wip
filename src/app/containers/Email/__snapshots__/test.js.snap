// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailContainer /> renders without crashing 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="email"
    >
      Next enter your email
    </label>
    <div
      className="control "
    >
      <input
        aria-describedby="email"
        className="input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        required={true}
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="control"
    >
      <button
        className="button button is-large is-primary"
      >
        <span>
          Continue
        </span>
      </button>
    </div>
  </div>
</form>
`;
